DEVPROD-3946: add Honeycomb attributes for spawn host modification jobs

Kimchelly opened the pull request:
DEVPROD-3946

### Description
Attach more spawn host attributes to the Amboy job Honeycomb spans (specifically for for spawn host modification jobs). This lets us do things like set up Honeycomb dashboards for stops/starts and filter by host fields like whether it's stopping an unexpirable host, the instance type, the source of the stop/start action (a user pressing the button vs. the sleep schedule), etc.

If the job errors (e.g. because AWS returns an error while stopping the host), Amboy's Honeycomb setup already records those errors and stack traces for us. They're also already logged to Splunk.

### Testing
Stopped/starting a spawn host in staging and verified that [the Honeycomb data is being sent](https://ui.honeycomb.io/mongodb-4b/environments/staging/datasets/evergreen/result/cSoZWBv2MFb) and is queryable.

### Documentation
N/A

hadjri reviewed (changes requested):
Changes LGTM - can we add a couple lines to the spawnhost modify/stop/terminate/start unit tests to check that the source is set as we expect it to? Also we should update the PR title to be descriptive.

Kimchelly commented:
Updated the title and added some tests - nice catch btw! it turns out I actually forgot to set the type for the global constants and didn't notice until the tests failed.
